daml-script/export/src/main/scala/com/daml/script/export/Main.scala

```diff
@@ -45,29 +45,35 @@ object Main {
       ec: ExecutionContext,
       esf: ExecutionSequencerFactory,
       mat: Materializer,
-  ): Future[Unit] =
-    for {
-      client <- LedgerClient.singleHost(config.ledgerHost, config.ledgerPort, clientConfig)
-      acs <- LedgerUtils.getACS(client, config.parties, config.start)
-      trees <- LedgerUtils.getTransactionTrees(client, config.parties, config.start, config.end)
-      acsPkgRefs = TreeUtils.contractsReferences(acs.values)
-      treePkgRefs = TreeUtils.treesReferences(trees)
-      pkgRefs = acsPkgRefs ++ treePkgRefs
-      pkgs <- Dependencies.fetchPackages(client, pkgRefs.toList)
-      _ = Dump.writeDump(
-        config.sdkVersion,
-        config.damlScriptLib,
-        config.outputPath,
-        acs,
-        trees,
-        pkgRefs,
-        pkgs,
-        config.acsBatchSize,
-      )
-    } yield ()
+  ): Future[Unit] = {
+    config.exportType match {
+      case Some(exportScript: ExportScript) =>
+        for {
+          client <- LedgerClient.singleHost(config.ledgerHost, config.ledgerPort, clientConfig)
+          acs <- LedgerUtils.getACS(client, config.parties, config.start)
+          trees <- LedgerUtils.getTransactionTrees(client, config.parties, config.start, config.end)
+          acsPkgRefs = TreeUtils.contractsReferences(acs.values)
+          treePkgRefs = TreeUtils.treesReferences(trees)
+          pkgRefs = acsPkgRefs ++ treePkgRefs
+          pkgs <- Dependencies.fetchPackages(client, pkgRefs.toList)
+          _ = Export.writeExport(
+            exportScript.sdkVersion,
+            exportScript.damlScriptLib,
+            exportScript.outputPath,
+            acs,
+            trees,
+            pkgRefs,
+            pkgs,
+            exportScript.acsBatchSize,
+          )
+        } yield ()
```

Comment:
```suggestion
        } {
          Export.writeExport(
            exportScript.sdkVersion,
            exportScript.damlScriptLib,
            exportScript.outputPath,
            acs,
            trees,
            pkgRefs,
            pkgs,
            exportScript.acsBatchSize,
          )
        }
```
Removing the `yield` is equivalent to turning the final `map` in a `foreach`. Makes sense?

Reply:
I may be misunderstanding, but this doesn't type-check. The function's return type is `Future[Unit]`, but without the `yield` this returns `Unit`.